### ISSUE TITLE
Unable to save active section plane to a scene using the SketchUp C API

### ISSUE THREAD

**Issue by Whaat** [SketchUp/api-issue-tracker#1015]
Using current version of SketchUp C API as of today (Nov 25, 2024).  SDK_WIN_x64_2024-0-594 - Windows 11

It does not seem possible to save the active section plane to a scene using the C API.  I have tried doing it multiple ways (order of operations) and it never works so I suspect a bug.

I HAVE been able to save camera properties to a scene using the C API but using a similar approach does not work when saving an active section to a scene.

The test program is a simple modification of one of the sample projects 'WritingToAskpFile'.  

In my test program, I create two section cuts and two scenes.  The expected result is that one scene will have an active cut and the other scene will have the other active cut.

But the actual result is that both scenes have NO active section cut. The sections are created but are not made active when the scene is activated.

See attached main.cpp which is a simple modification to the sample project  'WritingToAskpFile'.   You should be able to recreate by opening the sample project and replacing main.cpp with the attached version.

The output skp created is called 'new_model.skp'.  Inspect this file after running the program to see the results which will show that when clicking on scene, there is no active section cut.

![image](https://github.com/user-attachments/assets/9e6b5073-954a-4da2-96f5-cc84edbd0ed5)

This bug is a problem for me because I am working on a feature to export components as individual SKP files and then automatically add section cuts / scenes to the SKP using the C API so I can then reference these SKP files / models in a Layout Document that is generated using the Layout Ruby API.  This bug is preventing me from being able to implement this feature.

[main.zip](https://github.com/user-attachments/files/17905624/main.zip)

**Comment by thomthom** [comment]
@Whaat - just making sure I understand the issue correctly. If the C API had something similar to to Ruby's `page.update(flags)` - that would work, right?

Something like:
```cpp
SUSceneUpdate(scene_ref, SU_SCENE_UPDATE_SECTION_PLANES | SU_SCENE_UPDATE_CAMERA);
```

?

**Comment by Whaat** [comment]
Yes, that would be ideal.  However, as I said, it is possible to set the Camera to the scene using the currently available C API calls.  But doing the equivalent steps for section planes is not working.

So I don't think a new function needs to be created.  Just need to fix the current ones.

**Comment by thomthom** [comment]
Setting active section planes per scene is a little bit more tricky than camera. There is only one camera per scene. But in a model there can be multiple active section cuts in various entities collections.

Setting section planes as active throughout the model and then doing an "update" seems to be a simple way to allow this ability.

**Comment by thomthom** [comment]
@Whaat - I just realized that `SUModelAddScenes` will also do an "update" - same as what my proposed `SUSceneUpdate` would do.

But I also now understand why your code isn't working. We realized last week what `SUSceneCreate` creates a scene that have none of the `SUSceneFlags` set. (https://extensions.sketchup.com/developers/sketchup_c_api/sketchup/scene_8h.html#SUSceneFlags)

You can observe this if you open a model create from your script and inspect the Scene properties in the UI.

It's a poor API design. Something we could risk breaking existing users if we correct now. But we've already updated the documentation internally to clarify how to create a scene that will remember all properties, similar to what the UI and the Ruby API does by default:

```
SUSceneRef scene = SU_INVALID;
SUSceneCreate(&scene);
SUSceneSetFlags(scene, FLAG_USE_ALL);
SUSceneSetName(scene, "Hello World");
SUModelAddScenes(model, 1, &scene);
```

**Comment by Whaat** [comment]
@thomthom You guys must have made a code change in SU2025 because the behavior is different now.

I haven't tested section planes yet but setting the camera for a scene works differently in SU2025 vs SU2024.

In SU2025, I can use this sequence to set the camera for the new scene.
```
SUCameraRef cam = createCamera(modelBounds, direction);
SUModelSetCamera(model, &cam);

SUSceneRef scene = SU_INVALID;
SUResult res = SUSceneCreate(&scene);
SUSceneSetFlags(scene, FLAG_USE_ALL);
SUSceneSetName(scene, name);
res = SUModelAddScenes(model, 1, &scene);
```

But this same logic does NOT work in SU2024.

In SU2024, I had to do this:
```
SUSceneRef scene = SU_INVALID;
SUResult res = SUSceneCreate(&scene);
SUSceneSetName(scene, name);
// More code here for creating the SUCameraRef, setting the properties, calling SUSceneSetCamera and then SUSceneSetUseCamera
res = SUModelAddScenes(model, 1, &scene);
```

Is there a snippet of code that will work to set up the scenes for both SU2024 and SU2025?

**Comment by Whaat** [comment]
For now, I've decided to do a SU version check and use different logic based on whether the version is SU2025 or newer

**Comment by thomthom** [comment]
@Whaat - we found issues with the C API Scenes prior to SU2025.0. `SUSceneSetFlags` could set the model into a state where it would crash. And it was inconsistency in regard to when "Update" was done. For instance, `SUModelAddScene` and `SUModelAddScenes` did not both call "Update".

We made corrections and unifications to the SU2025.0 API and updated documentation to clearer examples on how to use it.
